modules/neovim/init/spellcheck.nix

```diff
@@ -58,18 +96,53 @@ in {
   };
 
   config = mkIf cfg.enable {
-    vim.luaConfigRC.spellcheck = entryAfter ["basic"] ''
-      vim.opt.spell = true
-      vim.opt.spelllang = ${listToLuaTable cfg.languages}
-
-      -- Disable spellchecking for certain filetypes
-      -- as configured by `vim.spellcheck.ignoredFiletypes`
-      vim.api.nvim_create_autocmd({ "FileType" }, {
-        pattern = ${listToLuaTable cfg.ignoredFiletypes},
-        callback = function()
-          vim.opt_local.spell = false
-        end,
-      })
-    '';
+    vim = {
+      additionalRuntimePaths = let
+        spellfilesJoined = pkgs.symlinkJoin {
+          name = "nvf-spellfiles-joined";
+          paths = mapAttrsToList (name: value: pkgs.writeTextDir "spell/${name}.add" (concatStringsSep "\n" value)) cfg.extraSpellFiles;
+          postBuild = ''
+            echo "Spellfiles joined"
+          '';
```

Comment:
Is the echo really necessary?

Also, just do the compilation here instead of in a separate derivation.

Reply:
Necessary? No. But I prefer displaying build phases as they happen.